harmony/harmonyparser.go

```diff
@@ -75,20 +88,29 @@ func (s *HarmonyParser) AddImplicitStart() {
 	s.acc.WriteString("<|start|>assistant")
 }
 
-func (s *HarmonyParser) AddImplicitStartOrPrefill(lastMessage *api.Message) {
-	if lastMessage != nil && lastMessage.Role == "assistant" {
-		// handle prefilling conditions
-		if lastMessage.Content != "" {
-			s.acc.WriteString("<|start|>assistant<|channel|>final<|message|>")
-			return
-		} else if lastMessage.Thinking != "" {
-			s.acc.WriteString("<|start|>assistant<|channel|>analysis<|message|>")
-			return
-		}
+// AddImplicitStartWithPrefill writes the provided prefill string to the accumulator
```

Comment:
as written this should still be called `AddImplicitStartOrPrefill` because the implicit start is still added if prefill is empty (which is actually the more common case).